Makefile

```diff
@@ -5,6 +5,13 @@ NOSTRETCH ?= 0
 NOBUSTER ?= 0
 NOBULLSEYE ?= 0
 
+export Q ?= @
+ifneq ($(QUIET),)
+override Q:=
+endif
+SONIC_OVERRIDE_BUILD_VARS += Q='$(Q)'
+export SONIC_OVERRIDE_BUILD_VARS
```

Comment:
What's the purpose of `SONIC_OVERRIDE_BUILD_VARS`? What do you mean by setting make options dynamically?

Reply:
In Makefile.work, SONIC_OVERRIDE_BUILD_VARS flag is already passed as a dynamic argument to the slave build container. 
      ```
SONIC_BUILD_INSTRUCTION :=  make \
                           -f slave.mk \
                              .....
                            $(SONIC_SLAVE_DOCKER_DRIVER=$(SONIC_SLAVE_DOCKER_DRIVER) \
                            **$(SONIC_OVERRIDE_BUILD_VARS)**
```

So setting any value to this variable will become an argument to the slave build container. As the name suggests, any make variable can be set dynamically. 

Already so many variables are being already passed as individual arguments. So the SONIC_OVERRIDE_BUILD_VARS variable will eliminate the future addition to it.